packages/react-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -51,6 +51,8 @@ export interface NumberInputProps extends React.HTMLProps<HTMLDivElement> {
   minusBtnProps?: ButtonProps;
   /** Additional properties added to the plus button */
   plusBtnProps?: ButtonProps;
+  /** Allow the user to clear out the input to an empty string (recommended) */
+  allowEmptyInput?: boolean;
```

Comment:
Should this be marked as beta?

Reply:
perhaps.. or maybe even deprecated as I believe we intend to remove it in v5 and have it as the default behavior.

@tlabaj or @nicolethoen input?

Reply:
I do not think deprecating it would be technically correct here especially since it is the recommend approach.  We could mark it beta... But I actually don't think it is worthwhile since we will remove it with v5 and this is the last v4 release we have planned.

Reply:
We should open up a follow up issue to track. All the work on the main branch will be merged back into the v5 branch at the end of the milestone.

Reply:
Opened an issue: https://github.com/patternfly/patternfly-react/issues/8526